src/main/java/io/debezium/connector/yugabytedb/YugabyteDBSnapshotChangeEventSource.java

```diff
@@ -805,6 +830,28 @@ public void markSnapshotDoneOnServer(YBPartition partition, YugabyteDBOffsetCont
 
   }
 
+  /**
+   * Last snapshot record of the last snapshot batch will have a unique marker that will be used for indentification.
+   * Set all the fields of the record's Checkpoint to max values and set snapshot key to 'LAST_SNAPSHOT_RECORD'
+   * @param recordCheckpoint the record's checkpoint of the last snapshot record
+   */
+  private void setIdentificationMarkerForLastSnapshotRecord(OpId recordCheckpoint) {
+    recordCheckpoint.setTerm(Long.MAX_VALUE);
```

Comment:
Rather than using setters for the values, create a new `OpId` object and assign it to `recordCheckpoint` - this will ensure that the `OpId` object remains read-only consistently and we do not accidentally end up modifying the values.

Reply:
This wont change the actual object `recordCheckpoint` is pointing to. We, ultimately, pass `recordCheckpoint` to kafka.

Reply:
This change has been done.